### ISSUE TITLE
LookUp support

### ISSUE THREAD

**Issue by TonyTroeff** [microsoft/Power-Fx#106]
Hello, I am experimenting with this library and encountered the following problem:
This is my code:
```
var expression = "LookUp(Settings, Name = \"Client\")";
var engine = new RecalcEngine();
var parameters = FormulaValue.RecordFromFields(
    new NamedValue("Settings",
        FormulaValue.TableFromRecords(
            FormulaValue.RecordFromFields(new NamedValue("Name", FormulaValue.New("Client"))),
            FormulaValue.RecordFromFields(new NamedValue("Name", FormulaValue.New("Test"))))));
var result = engine.Eval(expression, parameters);
```
I expect that this expression will be evaluated correctly and will return only the first table record (the name of which is exactly "Client"). But instead an exception is thrown. This is its message:
```
Errors: Error 0-33: Invocation of unknown or unsupported function.
Error 17-21: Name isn't valid. This identifier isn't recognized.
Error 22-23: Incompatible types for comparison. These types can't be compared: Error, Text.
```

I found that in your `BuiltInFunctionsCore` file there are `public static readonly` fields for each built-in function but the `LookUp` and `Search` are missing. Maybe this is one reason for this behavior?

**Comment by lesaltzm** [comment]
Right. A bunch of the functions present in Canvas' PowerFx aren't yet implemented in this runtime. We're working to close that gap over time.

**Comment by MikeStall** [comment]
We should give a better error than "Name isn't valid. This identifier isn't recognized.". We should call out the specific identifier. 

@TonyTroeff - First and Filter are implemented, so you could use those to do the Lookup. 
Note that we don't have delegation implemented in the interpreter, so this would be a linear scan.

**Comment by timnudecision** [comment]
related:  ShowColumns is one of those not yet implemented.  it is a priority for my project, should that affect the roadmap.

is there a work-around?

**Comment by MikeStall** [comment]
@timnudecision -  That's correct - it's not implemented yet. can you file a separate issue just for that?    

You may be able to work around it via a ForAll. 
`ShowColumns(_table, "a")`

is the same as: 
`ForAll(_table, { a : a })`

**Comment by timnudecision** [comment]
that workaround works on the REPL and CanvasApp
thanks

**Comment by MikeStall** [comment]
Function is added.
